smf/nexus/method_script.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -x
+set -o errexit
+set -o pipefail
+
+. /lib/svc/share/smf_include.sh
+
+LISTEN_ADDR="$(svcprop -c -p config/listen_addr "${SMF_FMRI}")"
+DATALINK="$(svcprop -c -p config/datalink "${SMF_FMRI}")"
+OPTE_INTERFACE="$(svcprop -c -p config/opte_interface "${SMF_FMRI}")"
+OPTE_GATEWAY="$(svcprop -c -p config/opte_gateway "${SMF_FMRI}")"
+GATEWAY="$(svcprop -c -p config/gateway "${SMF_FMRI}")"
+
+if [[ $DATALINK == unknown ]] || [[ $GATEWAY == unknown ]]; then
+    printf 'ERROR: missing datalink or gateway\n' >&2
+    exit "$SMF_EXIT_ERR_CONFIG"
+fi
+
+# Set up underlay
+ipadm show-addr "$DATALINK/ll" || ipadm create-addr -t -T addrconf "$DATALINK/ll"
+ipadm show-addr "$DATALINK/omicron6"  || ipadm create-addr -t -T static -a "$LISTEN_ADDR" "$DATALINK/omicron6"
+route get -inet6 default -inet6 "$GATEWAY" || route add -inet6 default -inet6 "$GATEWAY"
+
+# Set up OPTE interface
+if [[ "$OPTE_GATEWAY" =~ .*:.* ]]; then
+    # IPv6 gateway
+    echo "IPv6 OPTE gateways are not yet supported"
+    exit 1
```

Comment:
This still needs to be patched; I don't understand how this is working today.

Reply:
Yea, IPv6 support for OPTE ports hasn't been fully fleshed out. Means only IPv4 addresses supported for services as well rn.